Issue #1511: Const aggregate Combine

hawkfish opened the pull request:
The Combine operation for aggregation must not destroy the source `STATE` because the window segment tree uses it  to combine tree levels. If an aggregate destroys (even by moving) that state, it will make future results invalid.

Commit:
Issue #1511: Aggregate Combine const

The Combine operation for aggregation
must not destroy the source STATE
because the window segment tree uses it
to combine tree levels. If an aggregate
destroys (even by copying) that state,
it will make the results invalid.

Commit:
Issue #1511: Fix self assignment

It appears that tidy only understands one
type of guard...

Commit:
Issue #1511: Aggregate Combine const

Disable incorrect check.

Commit:
Merge branch 'master' into hawkfish-combine-const

Mytherin commented on src/function/aggregate/distributive/entropy.cpp:
```diff
@@ -9,8 +9,23 @@ namespace duckdb {
+	DistinctMap *distinct;
+
+	// NOLINTNEXTLINE(bugprone-unhandled-self-assignment)
+	EntropyState &operator=(const EntropyState &other) {
```
Perhaps use an explicit `Copy` function, so it is clear we are making an (expensive) copy and we can avoid the linter exception?

Mytherin replied:
It's always the same with those darn computers isn't it :)

Mytherin reviewed (commented):
Thanks for the PR! Looks great. Minor nitpick:

Commit:
Issue #1511: Const aggregate Combine

Replace assignment with named method to avoid
incomprehensible clang-tidy whinging.
Also delete assignment to prevent Bad Things.
